src/parseChangelog.ts

```diff
@@ -35,24 +39,31 @@ function parseChangelog({ changelogContent, repoUrl }) {
     unreleasedLinkReferenceDefinition,
   );
 
-  let mostRecentRelease;
-  let mostRecentCategory;
-  let currentChangeEntry;
+  let mostRecentRelease: string;
+  let mostRecentCategory: ChangeCategory | undefined | null;
+  let currentChangeEntry: string | undefined;
 
   /**
    * Finalize a change entry, adding it to the changelog.
    *
    * This is required because change entries can span multiple lines.
    *
-   * @param {Object} [options]
-   * @param {boolean} [options.removeTrailingNewline] - Indicates that the
-   *   trailing newline is not a part of the change description, so should be
-   *   removed.
+   * @param options
+   * @param options.removeTrailingNewline - Indicates that the trailing newline
+   * is not a part of the change description, and should therefore be removed.
    */
-  function finalizePreviousChange({ removeTrailingNewline = false } = {}) {
+  function finalizePreviousChange({
+    removeTrailingNewline = false,
+  }: {
+    removeTrailingNewline?: boolean;
+  } = {}) {
     if (!currentChangeEntry) {
       return;
     }
+    /* istanbul ignore next: This doesn't happen in practice. */
```

Comment:
I think this could happen with two successive, empty change categories/releases? :thinking: